components/api/core/response_handler.bal

```diff
@@ -26,18 +26,18 @@ public function buildErrorResponse (int code, string message, string description
     var errJson = json.convert(errPassed);
     if (errJson is json) {
         res.setJsonPayload(errJson);
-        res.statusCode = http:INTERNAL_SERVER_ERROR_500;
+        res.statusCode = statusCode;
     } else {
-        res = buildUnknownErrorResponse("Error occured when converting Error struct to Json");
+        res = buildUnknownErrorResponse();
     }
     return res; 
 }
 
-function buildUnknownErrorResponse (string msg) returns http:Response {
+function buildUnknownErrorResponse () returns http:Response {
     http:Response res = new;
     json errDefault = {
-        code: API_DEFAULT_STATUSCODE,
-        message: msg,
+        code: API_ERROR_CODE,
+        message: "Error occured when converting Error struct to Json",
```

Comment:
Let's change this to something more meaningful. The conversion error is not something that would be useful to an API consumer.

Suggestion - Unexpected error occurred